test/unit/controllers/secret_recovery.controller.spec.ts

```diff
@@ -111,6 +111,19 @@ describe("SecretRecoveryController", () => {
                 expect(response.body.type).toBe("SECRET");
             });
     })
+
+    it("downloads secret value", async () => {
+        const app = setupApp(SecretRecoveryController, mockForDownload);
+        await request(app)
+            .put(`/api/secret-recovery/${ REQUEST_ID }/download-secret`)
+            .send({ challenge: CHALLENGE })
+            .expect(200)
+            .then(response => {
+                expect(response.body.value).toBe(SECRET_VALUE);
+            });
+        secretRecoveryRequest.verify(instance => instance.markDownloaded(It.IsAny(), CHALLENGE, true));
+        secretRecoveryRequestRepository.verify(instance => instance.save(secretRecoveryRequest.object()));
+    })
 })
```

Comment:
Just to be 100% complete II would also mock and test the failure case using something like:
`secretRecoveryRequest.setup(instance => instance.markDownloaded(It.IsAny(), It.IsAny(), true)).throws("Error")`

Reply:
Done, the resource then returns a 400.